grpc/src/test/java/com/linecorp/armeria/it/grpc/GrpcHttpJsonTranscodingServiceAnnotatedAuthServiceTest.java

```diff
@@ -0,0 +1,154 @@
+import com.linecorp.armeria.server.logging.LoggingService;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import io.grpc.Metadata;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.MetadataUtils;
+import io.grpc.stub.StreamObserver;
+
+public class GrpcHttpJsonTranscodingServiceAnnotatedAuthServiceTest {
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            final GrpcService grpcService = GrpcService.builder()
+                    .addService(new AuthenticatedHttpJsonTranscodingTestService())
+                    .enableHttpJsonTranscoding(true)
+                    .build();
+            sb.requestTimeoutMillis(5000);
+            sb.decorator(LoggingService.newDecorator());
+            sb.service(grpcService);
+        }
+    };
+
+    private static final String TEST_CREDENTIAL_KEY = "credential";
+
+    private final ObjectMapper mapper = JacksonUtil.newDefaultObjectMapper();
+
+    private final BlockingWebClient webClient = server.webClient().blocking();
+
+    private final HttpJsonTranscodingTestServiceBlockingStub grpcClient =
+            GrpcClients.newClient(server.httpUri(GrpcSerializationFormats.PROTO),
+                    HttpJsonTranscodingTestServiceBlockingStub.class);
+
+    @Test
+    void testAuthenticatedRpcMethod() throws Exception {
+        final Transcoding.GetMessageRequestV1 requestMessage = Transcoding.GetMessageRequestV1.newBuilder()
+                .setName("messages/1").build();
+        final Throwable exception = assertThrows(Throwable.class,
+                () -> grpcClient.getMessageV1(requestMessage).getText());
+        assertThat(exception).isInstanceOf(StatusRuntimeException.class);
+        assertThat(((StatusRuntimeException) exception).getStatus().getCode())
+                .isEqualTo(Status.UNAUTHENTICATED.getCode());
+
+        final Metadata metadata = new Metadata();
+        metadata.put(Metadata.Key.of(TEST_CREDENTIAL_KEY, Metadata.ASCII_STRING_MARSHALLER),
+                "some-credential-string");
+        final Transcoding.Message result = grpcClient.withInterceptors(
+                MetadataUtils.newAttachHeadersInterceptor(metadata)
+        ).getMessageV1(requestMessage);
+        assertThat(result.getText()).isEqualTo("messages/1");
+    }
+
+    @Test
+    void testAuthenticatedHttpJsonTranscoding() throws Exception {
+        final AggregatedHttpResponse failResponse = webClient.get("/v1/messages/1");
+        assertThat(failResponse.status()).isEqualTo(HttpStatus.UNAUTHORIZED);
+
+        final AggregatedHttpResponse successResponse = webClient
+                .execute(RequestHeaders.of(HttpMethod.GET, "/v1/messages/1",
+                        TEST_CREDENTIAL_KEY, "some-credential-string"));
+        final JsonNode root = mapper.readTree(successResponse.contentUtf8());
```

Comment:
nit: Use `asJson(JsonNode.class)`?
```suggestion
        final JsonNode root = 
            webClient.prepare()
                     .get("/v1/messages/1")
                     .header(TEST_CREDENTIAL_KEY, "some-credential-string")
                     .asJson(JsonNode.class)
                     .execute()
                     .content();
```

Reply:
I see. You can ignore this comment. I will clean up our test code later.